### PR TITLE
Custom plotter for plotting bandstructures

### PR DESCRIPTION
**Pull request by mfherbst**

### CODE REVIEW

File: src/common/units.jl [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    K  = 3.166810508e-6      # Kelvin -> Hartree
+    Ry = 0.5                 # Rydberg -> Hartree
+end
+
+"""
+    unit(symbol)
+Get the factor converting from the unit `symbol` to atomic units.
+E.g. `unit(:eV)` returns the conversion factor from electron volts to Hartree.
+"""
+unit_to_au(sym::Symbol) = getfield(units, sym)
```
Comment (antoine-levitt) [inline src/common/units.jl]: Inconsistent with docstring (see why I hate those? :p)

File: examples/gross_pitaevskii.jl [hunk 2]
```diff
@@ -5,6 +5,7 @@ using PyCall
 using DFTK
 using Printf
 using LinearAlgebra
+using PyPlot
```
Comment (antoine-levitt) [inline examples/gross_pitaevskii.jl]: Do we want to go full Plots and drop PyPlot everywhere?

File: src/postprocess/band_structure.jl [hunk 3]
```diff
@@ -21,28 +22,81 @@ function high_symmetry_kpath(basis, kline_density)
...
-    bs = pymatgen_bandstructure(basis, band_data, klabels, fermi_level=εF)
-    bsplot = plotter.BSPlotter(bs)
-    plt = bsplot.get_plot()
-    plt.autoscale()
-    plt
+    data = plotter.BSPlotter(bs).bs_plot_data(zero_to_efermi=false)
+
+    # Collect some useful info in a more useful way
+    spins = [("1", :blue)]
+    bs.is_spin_polarized && (spins = [("1", :blue), ("-1", :red)])
```
Comment (antoine-levitt) [inline src/postprocess/band_structure.jl]: Might as well do an if

**Comment by antoine-levitt** [comment]
Cool! The conflicts with termpocalypse are building up and it's making me a bit uneasy, though...

**Comment by mfherbst** [comment]
I know, but trust in git ;), also I really want to get this stuff working first before I loose a day for merging.

File: src/common/units.jl (hunk 1 above)
Comment (mfherbst) [reply src/common/units.jl]: :smile:

File: examples/gross_pitaevskii.jl (hunk 2 above)
Comment (mfherbst) [reply examples/gross_pitaevskii.jl]: Yes, I'm on that. There are a few tiny things though, which came up related to labels taking real parts, etc.

File: src/postprocess/band_structure.jl (hunk 3 above)
Comment (mfherbst) [reply src/postprocess/band_structure.jl]: ok.

File: examples/gross_pitaevskii.jl (hunk 2 above)
Comment (mfherbst) [reply examples/gross_pitaevskii.jl]: It's really annoying that the default GR backend does not support unicode, actually.

**Pull request merged**